tests/minifier.js

```diff
@@ -1690,6 +1690,21 @@ QUnit.test('phrasing content', function(assert) {
   assert.equal(minify(input, { html5: true }), input);
 });
 
+// https://github.com/kangax/html-minifier/issues/888
+QUnit.test('ul/ol should be phrasing content', function(assert) {
+  var input, output;
+
+  input = '<p>a<ul><li>item</li></ul>';
+  output = '<p>a</p><ul><li>item</li></ul>';
+  assert.equal(minify(input, { html5: true }), output);
+  output = '<p>a<ul><li>item</li></ul></p>';
+  assert.equal(minify(input, { html5: false }), output);
+
+  input = '<p>a<ol><li>item</li></ol></p>';
+  output = '<p>a</p><ol><li>item</li></ol><p></p>';
+  assert.equal(minify(input, { html5: true }), output);
```

Comment:
Tests look good, thanks.

Can you add one more that also does `removeOptionalTags: true` to make sure the newly placed `</p>` doesn't get removed.

Reply:
The generated </p> before ul/ol should be removed because it's optional. I also added test to check the <p></p> at the end.

Reply:
Perfect - thanks! :+1: